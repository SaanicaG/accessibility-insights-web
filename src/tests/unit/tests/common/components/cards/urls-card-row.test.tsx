// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { render } from '@testing-library/react';
import { UrlsCardRow, UrlsCardRowProps } from 'common/components/cards/urls-card-row';
import { CardRowDeps } from 'common/configs/unified-result-property-configurations';
import * as React from 'react';
import { NewTabLinkWithConfirmationDialog } from 'reports/components/new-tab-link-confirmation-dialog';

describe('UrlsCardRow', () => {
    it('renders with string-only URLs', () => {
        const props: UrlsCardRowProps = {
            propertyData: {
                urlInfos: [
                    { url: 'https://www.test.com', baselineStatus: 'unknown' },
                    { url: 'https://www.test.com/more/tests', baselineStatus: 'unknown' },
                ],
            },
            deps: {
                LinkComponent: NewTabLinkWithConfirmationDialog,
            } as CardRowDeps,
            index: -1,
        };
        const renderResult = render(<UrlsCardRow {...props} />);
        expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it('renders with baseline-aware URLs', () => {
        const props: UrlsCardRowProps = {
            propertyData: {
                urlInfos: [
                    { url: 'https://www.test.com', baselineStatus: 'unknown' },
                    { url: 'https://www.test.com/more/tests', baselineStatus: 'existing' },
                    { url: 'https://www.test.com/still/more/tests', baselineStatus: 'new' },
                ],
            },
            deps: {
                LinkComponent: NewTabLinkWithConfirmationDialog,
            } as CardRowDeps,
            index: -1,
        };
        const renderResult = render(<UrlsCardRow {...props} />);
        expect(renderResult.asFragment()).toMatchSnapshot();
    });
});
