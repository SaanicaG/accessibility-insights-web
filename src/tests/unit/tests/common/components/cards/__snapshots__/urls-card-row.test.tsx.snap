// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlsCardRow renders with baseline-aware URLs 1`] = `
<DocumentFragment>
  <mock-simplecardrow
    content="[object Object]"
    label="URL"
    rowkey="urls-row--1"
  />
</DocumentFragment>
`;

exports[`UrlsCardRow renders with string-only URLs 1`] = `
<DocumentFragment>
  <mock-simplecardrow
    content="[object Object]"
    label="URL"
    rowkey="urls-row--1"
  />
</DocumentFragment>
`;
